migrations/000001_init_schema.up.sql

```diff
@@ -0,0 +1,32 @@
+CREATE TABLE IF NOT EXISTS users (
+    id uuid NOT NULL DEFAULT uuid_generate_v1(),
+    username varchar(512) NOT NULL UNIQUE,
+    PRIMARY KEY (id)
+);
+
+CREATE TABLE IF NOT EXISTS notification_types (
+    id UUID NOT NULL DEFAULT uuid_generate_v1(),
+    name varchar(32) NOT NULL UNIQUE,
+    PRIMARY KEY (id)
+);
+
+CREATE TABLE IF NOT EXISTS notifications (
+    id UUID DEFAULT uuid_generate_v1(),
+    notification_type_id uuid NOT NULL REFERENCES notification_types(id),
+    user_id uuid NOT NULL REFERENCES users(id) ON DELETE CASCADE,
+    subject text NOT NULL,
+    seen boolean NOT NULL DEFAULT false,
+    deleted boolean NOT NULL DEFAULT false,
+    time_created timestamp with time zone NOT NULL DEFAULT now(),
```

Comment:
oh, timezones. heh.

so this may be deliberate on your part, but elsewhere in our code we tend to use `timestamp without time zone` and store timestamps in UTC-7 or whatever else is the server timezone. This is, of course, somewhat less than ideal, but I don't know if we want to keep it consistent. `timestamp with time zone`, as it happens, actually just stores a point in time UTC, and then translates it to the requestor's time zone (there's no actual time zone information stored in _either_ type, which is annoying and confusing in my opinion, but anyway). If we actually want to store time zones rather than points in time, then of course we can take that into account

anyway, this seems fine to me, but I want to make sure that if you don't know about the weirdness of timestamps in postgres and in our codebase, you have advance warning :D

Reply:
Thanks for the heads up. I think that a point in time in UTC will work, but it is good to know about some of the finer details.